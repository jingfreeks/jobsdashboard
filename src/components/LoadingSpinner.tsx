import React from 'react';

const LoadingSpinner: React.FC = () => {
  return (
    <div className="min-h-screen flex items-center justify-center bg-slate-100">
      <div className="flex flex-col items-center gap-4">
        <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"></div>
        <p className="text-slate-600 font-medium">Loading...</p>
      </div>
    </div>
  );
};

export default LoadingSpinner; 