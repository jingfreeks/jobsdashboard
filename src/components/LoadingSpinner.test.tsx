import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner', () => {
  it('renders loading spinner with text', () => {
    const { container } = render(<LoadingSpinner />);
    
    expect(screen.getByText('Loading...')).toBeInTheDocument();
    expect(container.querySelector('.animate-spin')).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });

  it('has correct styling classes', () => {
    const { container } = render(<LoadingSpinner />);
    
    const spinnerContainer = container.firstChild as HTMLElement;
    expect(spinnerContainer).toHaveClass('min-h-screen', 'flex', 'items-center', 'justify-center', 'bg-slate-100');
    
    const spinner = container.querySelector('.animate-spin');
    expect(spinner).toHaveClass('animate-spin', 'rounded-full', 'h-12', 'w-12', 'border-b-2', 'border-blue-600');
  });

  it('matches loading spinner snapshot', () => {
    const { container } = render(<LoadingSpinner />);
    expect(container).toMatchSnapshot();
  });
}); 