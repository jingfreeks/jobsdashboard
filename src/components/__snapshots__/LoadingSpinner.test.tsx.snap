// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner > matches loading spinner snapshot 1`] = `
<div>
  <div
    class="min-h-screen flex items-center justify-center bg-slate-100"
  >
    <div
      class="flex flex-col items-center gap-4"
    >
      <div
        class="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"
      />
      <p
        class="text-slate-600 font-medium"
      >
        Loading...
      </p>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner > renders loading spinner with text 1`] = `
<div>
  <div
    class="min-h-screen flex items-center justify-center bg-slate-100"
  >
    <div
      class="flex flex-col items-center gap-4"
    >
      <div
        class="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"
      />
      <p
        class="text-slate-600 font-medium"
      >
        Loading...
      </p>
    </div>
  </div>
</div>
`;
